File: badreq.py
from http.server import BaseHTTPRequestHandler, HTTPServer
from urllib.parse import urlparse, parse_qs

class InsecureHTTPRequestHandler(BaseHTTPRequestHandler):
    def do_GET(self):
        # Parse query parameters
        query_components = parse_qs(urlparse(self.path).query)
        
        # Get the 'name' parameter from the query string
        name = query_components.get('name', [''])[0]

        # Insecure: Reflect the 'name' parameter directly in the response without sanitizing it
        self.send_response(200)
        self.send_header('Content-type', 'text/html')
        self.end_headers()
        self.wfile.write(bytes(f"<h1>Hello, {name}</h1>", "utf-8"))

def run(server_class=HTTPServer, handler_class=InsecureHTTPRequestHandler, port=8080):
    server_address = ('', port)
    httpd = server_class(server_address, handler_class)
    print(f'Server running on port {port}...')
    httpd.serve_forever()

if __name__ == '__main__':
    run()

File: badreq.js
const http = require('http');
const url = require('url');

// Create an HTTP server
const server = http.createServer((req, res) => {
    const queryObject = url.parse(req.url, true).query;

    // Get the 'name' parameter from the query string
    const name = queryObject.name;

    // Insecure: Reflect the 'name' parameter directly in the response without sanitizing it
    res.writeHead(200, { 'Content-Type': 'text/html' });
    res.end(`<h1>Hello, ${name}</h1>`);
});

// Start the server
server.listen(3000, () => {
    console.log('Server is running on http://localhost:3000');
});
